src/Internal/Service/Blockfrost.purs

```diff
@@ -52,36 +57,63 @@ import Data.Bifunctor (lmap)
 import Data.Either (Either(Left, Right), note)
 import Data.Generic.Rep (class Generic)
 import Data.HTTP.Method (Method(GET, POST))
+import Data.Map (Map)
 import Data.Map as Map
-import Data.Maybe (Maybe, maybe)
+import Data.Maybe (Maybe(Just, Nothing), fromJust, maybe)
 import Data.MediaType (MediaType)
 import Data.Newtype (class Newtype, unwrap, wrap)
 import Data.Show.Generic (genericShow)
-import Data.Traversable (for)
+import Data.Traversable (for, traverse_)
+import Data.Tuple (Tuple(Tuple))
+import Data.Tuple.Nested (type (/\))
 import Effect.Aff (Aff)
 import Effect.Aff.Class (liftAff)
+import Effect.Class (liftEffect)
+import Effect.Ref (Ref)
+import Effect.Ref (modify_, new, read) as Ref
+import Partial.Unsafe (unsafePartial)
 import Undefined (undefined)
 
 --------------------------------------------------------------------------------
 -- BlockfrostServiceM
 --------------------------------------------------------------------------------
 
+type BlockfrostRawResponses = Maybe (Ref (Map BlockfrostEndpoint String))
```

Comment:
This wont work (well) for post requests, or endpoints whose results change over time. Maybe make it a multimap or just an array of pairs.

Reply:
Alternatively, switch to a hook instead of representing with data directly.

```purescript
type BlockfrostServiceParams =
  { ...
  , onBlockfrostRawGetResponse :: Maybe (BlockfrostEndpoint -> String -> Aff Unit)
  , onBlockfrostRawPostResponse :: Maybe (BlockfrostEndpoint -> MediaType -> Maybe Affjax.RequestBody -> String -> Aff Unit)
  }
```